### PR TITLE
Refactor test utils

### PR DESCRIPTION
**Pull request by zancas**
My goal with this work is to push test framework implementation details into a lower layer of abstraction, than test details.

Specifically, a test author should not need to write sleep or Duration or a polling loop in their integration test to ensure that the environment is correctly configured to test the properties of interest. That logic (e.g. syncing lightwalletd with the regtest blockchain, and syncing zingocli-client with lightwalletd, should be handled by standard reused utility functions UNLESS those things are being tested.

### CODE REVIEW

**Comment by zancas** [comment]
My goal with this work is to push test framework implementation details into a lower layer of abstraction, than test details.

Specifically, a test author should not need to write `sleep` or `duration` or a polling loop in their integration test to ensure that the environment is correctly configured to test the properties of interest.   That logic (e.g. syncing lightwalletd with the regtest blockchain, and syncing `zingocli-client` with `lightwalletd`, should be handled by standard reused utility functions _UNLESS_ those things are being tested.

File: cli/tests/integration_tests.rs [hunk 1]
```diff
@@ -110,26 +108,25 @@ fn send_mined_sapling_to_orchard() {
...
-        sleep(Duration::from_secs(2)).await;
-
-        client.do_sync(true).await.unwrap();
+        utils::increase_height_and_sync_client(&regtest_manager, &client, 2).await;
         let balance = client.do_balance().await;
         assert_eq!(balance["verified_orchard_balance"], 5000);
     });
 }
-
+/// This implements similar behavior to 'two_clients_a_coinbase_backed', but with the
```
Comment (AloeareV) [inline cli/tests/integration_tests.rs:116]: Stale comment

File: cli/tests/utils/mod.rs [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+use zingo_cli::regtest::RegtestManager;
+use zingolib::lightclient::LightClient;
+
+pub async fn increase_height_and_sync_client(
+    manager: &RegtestManager,
+    client: &LightClient,
+    n: u32,
+) {
+    let start_height = client
+        .do_wallet_last_scanned_height()
```
Comment (AloeareV) [inline cli/tests/utils/mod.rs:13]: If the client (or lightwalletd) isn't fully synced when this is called, then `start_height` will lead to an improperly determined target, which I think will depending on the numbers involved either early-return or hang forever.

File: cli/tests/integration_tests.rs (hunk 1 above)
Comment (zancas) [reply cli/tests/integration_tests.rs]: Fixed.

File: cli/tests/utils/mod.rs (hunk 2 above)
Comment (zancas) [reply cli/tests/utils/mod.rs:14]: I believe this is fixed by running `do_sync` immediately before this line.

Comment (zancas) [reply cli/tests/utils/mod.rs:14]: Though..  this fix assumes that `do_sync` fulfills its contract reliably.

**Review by AloeareV — APPROVED** [review]
Looks great!

**Pull request merged**